os/fs/smartfs/smartfs_utils.c

```diff
@@ -1375,10 +1349,25 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 		goto errout;
 	}
 
+	if (newchain) {
```

Comment:
I think this will work correct for mkdir.
But we should consider more for open->write->sync_internal.
for example we create /mnt/a as a 13th entry in /mnt folder, and write 300bytes and close, we have to follow below.
1. alloc sector for /mnt/a save sf regarding this. (but we do not asign alloc sector to sf->currsector, is it neede?)
2. fill the buffer when write being called.
3. Now we have to sync if close has called.
4. First write /mnt/a's firstsector (sf->currsector)
5. create 2nd entry sector of /mnt
6. Chain first~2nd entry of /mnt

Will you handle this next PR? otherwise, we need TODO

Reply:
Yes I need to make sf buffer changes and buffer sync modifications.
That is pending for smartfs_open